client/my-sites/plans-v2/utils.ts

```diff
@@ -94,13 +96,19 @@ export function productButtonLabel( product: SelectorProduct, isOwned: boolean )
 			: translate( 'Manage Subscription' );
 	}
 
-	return (
-		product.buttonLabel ??
-		translate( 'Get %s', {
-			args: product.displayName,
-			comment: '%s is the name of a product',
-		} )
-	);
+	const { buttonLabel, displayName } = product;
+
+	return buttonLabel ?? isString( displayName )
+		? translate( 'Get %s', {
+				args: displayName,
+				comment: '%s is the name of a product',
+		  } )
+		: translate( 'Get {{name/}}', {
+				components: {
+					name: displayName as ReactElement,
+				},
+				comment: '{{name/}} is the name of a product',
+		  } );
```

Comment:
Perhaps it can be simplified as `Get {{name/}}` with `name: <>{ displayName }</>`? That'd convert it to React regardless.

Reply:
Good catch!